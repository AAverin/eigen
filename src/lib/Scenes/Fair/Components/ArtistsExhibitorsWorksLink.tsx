import { Box, color, Flex, Sans, Separator, space } from "@artsy/palette"
import React from "react"
import { TouchableOpacity } from "react-native"
import styled from "styled-components/native"

interface Props {
  onViewAllExhibitorsPressed: () => void
  onViewAllArtworksPressed: () => void
  onViewAllArtistsPressed: () => void
}

// ts-lint:ignore
const TabWrapper = styled(Flex)`
  margin-top: ${space(1)};
  margin-bottom: ${space(2)};
  flex-direction: row;
  justify-content: space-evenly;
`
const LeftTab = styled(Box)`
  border-right-width: 1px;
  padding-right: 25px;
  border-right-color: ${color("black10")};
`

export class ArtistsExhibitorsWorksLink extends React.Component<Props> {
  render() {
    const { onViewAllExhibitorsPressed, onViewAllArtistsPressed, onViewAllArtworksPressed } = this.props
    return (
      <>
        <TabWrapper>
          <TouchableOpacity onPress={() => onViewAllArtistsPressed()}>
            <LeftTab>
              <Sans size="3">Artists</Sans>
            </LeftTab>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => onViewAllExhibitorsPressed()}>
            <LeftTab right="3">
              <Sans size="3">Exhibitors</Sans>
            </LeftTab>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => onViewAllArtworksPressed()}>
            <Box right="3">
              <Sans size="3">Works</Sans>
            </Box>
          </TouchableOpacity>
        </TabWrapper>
        <Separator mt="1" />
      </>
    )
  }
}
