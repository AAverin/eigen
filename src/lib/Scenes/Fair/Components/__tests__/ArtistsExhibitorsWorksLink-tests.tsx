import React from "react"
import * as renderer from "react-test-renderer"
import { ArtistsExhibitorsWorksLink } from "../ArtistsExhibitorsWorksLink"

const onViewAllExhibitorsPressed = jest.fn()
const data = {
  onViewAllExhibitorsPressed,
}
describe("ArtistsExhibitorsWorksLink", () => {
  it("Renders correctly", () => {
    const comp = renderer.create(<ArtistsExhibitorsWorksLink {...data as any} />)
    expect(comp).toMatchSnapshot()
  })
})
