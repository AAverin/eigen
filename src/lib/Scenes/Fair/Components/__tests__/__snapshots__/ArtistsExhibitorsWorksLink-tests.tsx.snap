// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsExhibitorsWorksLink Renders correctly 1`] = `
<View
  flexDirection="row"
  flexWrap="nowrap"
  justifyContent="space-between"
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "justifyContent": "space-between",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    align-items="base-line"
    flexDirection="row"
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="sc-bdVaJa bDWFJH"
      ellipsizeMode="tail"
      fontFamily="Unica77LL-Medium"
      fontSize="14px"
      lineHeight="20px"
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          },
          undefined,
        ]
      }
    >
      All artists
    </Text>
    <View
      ml={1}
      style={
        Array [
          Object {
            "marginLeft": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/horizontal_chevron.png",
          }
        }
        style={
          Object {
            "tintColor": "#C2C2C2",
            "top": 3,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    align-items="base-line"
    flexDirection="row"
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="sc-bdVaJa bDWFJH"
      ellipsizeMode="tail"
      fontFamily="Unica77LL-Medium"
      fontSize="14px"
      lineHeight="20px"
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          },
          undefined,
        ]
      }
    >
      All exhibitors
    </Text>
    <View
      ml={1}
      style={
        Array [
          Object {
            "marginLeft": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/horizontal_chevron.png",
          }
        }
        style={
          Object {
            "tintColor": "#C2C2C2",
            "top": 3,
          }
        }
      />
    </View>
  </View>
  <View
    accessible={true}
    align-items="base-line"
    flexDirection="row"
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="sc-bdVaJa bDWFJH"
      ellipsizeMode="tail"
      fontFamily="Unica77LL-Medium"
      fontSize="14px"
      lineHeight="20px"
      style={
        Array [
          Object {
            "fontFamily": "Unica77LL-Medium",
            "fontSize": 14,
            "lineHeight": 20,
          },
          undefined,
        ]
      }
    >
      All works
    </Text>
    <View
      ml={1}
      style={
        Array [
          Object {
            "marginLeft": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../images/horizontal_chevron.png",
          }
        }
        style={
          Object {
            "tintColor": "#C2C2C2",
            "top": 3,
          }
        }
      />
    </View>
  </View>
</View>
`;
